SingularityBase/src/main/java/com/hubspot/singularity/SingularityTaskIdHistory.java

```diff
@@ -15,17 +18,20 @@
   private final Optional<ExtendedTaskState> lastTaskState;
   private final Optional<String> runId;
 
+  private static final Comparator<SingularityTaskHistoryUpdate> comparator = new Comparator<SingularityTaskHistoryUpdate>() {
```

Comment:
Should this be composed with a comparator over timestamp, in case a task enters the same state more than once, as a tiebreaker?  or is that not possible?